lib/plugins/aws/invoke/index.js

```diff
@@ -24,43 +24,44 @@ class AwsInvoke {
   }
 
   extendedValidate() {
-    this.validate();
-
-    // validate function exists in service
-    this.options.functionObj = this.serverless.service.getFunction(this.options.function);
-    this.options.data = this.options.data || '';
-
-    return new BbPromise(resolve => {
-      if (this.options.data) {
-        resolve();
-      } else if (this.options.path) {
-        const absolutePath = path.isAbsolute(this.options.path) ?
-          this.options.path :
-          path.join(this.serverless.config.servicePath, this.options.path);
-        if (!this.serverless.utils.fileExistsSync(absolutePath)) {
-          throw new this.serverless.classes.Error('The file you provided does not exist.');
+    return this.validate().then(() => {
+      try {
+        // validate function exists in service
+        this.options.functionObj = this.serverless.service.getFunction(this.options.function);
+        this.options.data = this.options.data || '';
+      } catch (error) {
+        return BbPromise.reject(error);
```

Comment:
We're in `then` callback, so we should rather`throw` and `return` instead of `Promise.reject(...)` and `Promise.resolve(...)` (creating promises like that in `then` callback (or _async_ functions) is a coined anti-pattern).

Going further this `try/catch` wrap seems obsolete

Reply:
Hmm yes, I tried that but then the tests failed once again because for some reaon the thrown error is not translated to a Promise rejection...

Reply:
> the thrown error is not translated to a Promise rejection...

By promises design `throw` in all cases would translate into rejection. 

I can check what's the issue on that locally (I think it's worth to clean this up, as code will be turn more concise)